Dynamic Temperature Bound

nshah9856 opened the pull request:
Added the capability to dynamically bind upper and lower temperatures. 
The procedure followed is as follows:
Starting at 0, we decide to either increase or decrease the temperature depending on the LIMIT, explicitly set as HIGH_TEMP_LIMIT = 0.9999 and LOW_TEMP_LIMIT = 1e-10. Finally, we update the NUM_STEPS.

Commit:
Added capability to dynamically get lower and upper bound for temperatures

waynebhayes commented on src/methods/SANA.cpp:
```diff
@@ -1835,20 +1835,52 @@ uint SANA::getHighestIndex() const {
+			i++;
+		}
+	}
+	return LIMIT == HIGH_TEMP_LIMIT ? i+1 : i;
```
It'll almost never be *equal*; choose a < or > here.

nshah9856 replied:
Yes, I agree it won't be. But here the LIMIT just means the defined limits, either 0.9999 or 1e^-10. Following your psuedocode, if we are checking for High temperature limit, then we want to return i+1, else for Low temperature limit we want to return just i (since i is negative here, i+1 will change it in the wrong direction).

waynebhayes commented on src/methods/SANA.cpp:
```diff
@@ -1835,20 +1835,52 @@ uint SANA::getHighestIndex() const {
-#define LOG10_LOW_TEMP -10.0
-#define LOG10_HIGH_TEMP 10.0
-#define LOG10_NUM_STEPS 20 // looping through 0 <= i <= LOG10_NUM_STEPS
+#define HIGH_TEMP_LIMIT 0.99999
```
I know what you mean but these are actually pBad limits not temperature limits. Please rename appropriately.

nshah9856 replied:
Fixed!

waynebhayes commented on src/methods/SANA.cpp:
```diff
@@ -325,7 +325,7 @@ Alignment SANA::run() {
 	    double unweightdedScore = coreFreq[i][j]/(double)coreCount[i],
 		weightedScore = weightedCoreFreq[i][j]/totalCoreWeight[i];
-	    if(weightedScore > MIN_CORE_SCORE)printf("%s %s  %lu / %lu = %.6f weighted %.6f / %.6f = %.6f\n",
+	    if(weightedScore > MIN_CORE_SCORE)printf("%s %s  %lu / %lu = %.16f weighted %.16f / %.16f = %.16f\n",
```
Please leave these as %0.6f, not %.16f

nshah9856 replied:
Yea, this was updated by someone else idk how it got into our fork. But I reverted it back.

nshah9856 committed:
Update SANA.cpp

The change was made by someone else, but reverted back to the old code.

nshah9856 committed:
Update SANA.cpp

Changed the names from HIGH_TEMP_LIMIT to PBAD_HIGH_TEMP_LIMIT and LOW_TEMP_LIMIT to PBAD_LOW_TEMP_LIMIT to make it more readable and appropriate.

waynebhayes commented:
OK I see your point, but rather than hard-coding the assumption that
"limit" will be one of the two that are defined, instead let's pass in a
second parameter, a boolean "is_high", which is false if we're looking for
the lower limit and true if we're looking for the higher one.  Because
guarantede at some point somebody's going to call that function with a
value for limit you're not expecting, so force the user to choose which
endpoint it is when they call the function. (And modify your two calls
appropriately.)

waynebhayes commented on src/methods/SANA.cpp:
```diff
@@ -1835,20 +1835,52 @@ uint SANA::getHighestIndex() const {
+			i++;
+		}
+	}
+	return LIMIT == PBAD_HIGH_TEMP_LIMIT ? i+1 : i;
```
Please add a second parameter, "Boolean is_high", to the call to this function specifying if this is a high or a low temperature, and use that Boolean to make this decision. Don't assume any particular value for the LIMIT.

waynebhayes commented on src/methods/SANA.cpp:
```diff
@@ -1835,20 +1835,52 @@ uint SANA::getHighestIndex() const {
     //                      //otherwise my computer is very slow.
+	
+	cout << "Testing HIGH TEMP" << endl;
+	LOG10_HIGH_TEMP = temperatureBracket(PBAD_HIGH_TEMP_LIMIT);
```
Add second parameter here to this call, "true" for the high temp, and "false" for the lower one.

waynebhayes commented on src/methods/SANA.cpp:
```diff
@@ -1835,20 +1835,52 @@ uint SANA::getHighestIndex() const {
+	cout << "Testing HIGH TEMP" << endl;
+	LOG10_HIGH_TEMP = temperatureBracket(PBAD_HIGH_TEMP_LIMIT);
+	cout << "Testing LOW TEMP" << endl;
+	LOG10_LOW_TEMP = temperatureBracket(PBAD_LOW_TEMP_LIMIT);
```
Second parameter "false" here.

waynebhayes reviewed (changes requested):
See comments inline to add second parameter to the search function.

nshah9856 committed:
Added a second parameter to the temperatureBracket function to improve functionality of the function

nshah9856 commented:
Addressed your remarks regarding the second parameter needed in the temperatureBracket function. Sorry for the late change. I updated the .cpp and .hpp file to reflect the changes made.

waynebhayes commented:
Congratulations on your first accepted pull request. :-)

waynebhayes commented:
Cool, thanks.  Pull request acepted.  Congrats!

--
Wayne Hayes, Ph.D.
Associate Professor of Computer Science, University of California, Irvine
Director, UCI-SDSU Joint Ph.D. Program in Computational Science (UCI side)
*The evidence is now incontrovertible that Russia interfered with the 2016
US Presidential election by subtly manipulating public opinion via fake
social media accounts. Why? Follow the money: Trump has been laundering it
for Russian billionaires, including Putin, for 20 years. Details and fact
checking
at https://secure.avaaz.org/campaign/en/shocking_truth_about_trump/?bebhylb
<https://secure.avaaz.org/campaign/en/shocking_truth_about_trump/?bebhylb>.*
The opinions expressed in this email by Professor Hayes are his and his
alone, and are not meant to represent those of the University of California.

On Sat, Nov 24, 2018 at 5:20 PM, Nisarg Shah <notifications@github.com>
wrote:

> Addressed your remarks regarding the second parameter needed in the
> temperatureBracket function. Sorry for the late change. I changed the .cpp
> and .hpp file to reflect the changes made.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/waynebhayes/SANA/pull/47#issuecomment-441407986>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ATUkjAddTlxKQlMQR7hHVR436qc0l8CPks5uyfBmgaJpZM4YqZvL>
> .
>
